plugins/inputs/docker/docker.go

```diff
@@ -296,6 +311,22 @@ func (d *Docker) gatherContainer(
 		tags[k] = label
 	}
 
+	// Add whitelisted environment variables to tags
+	if len(d.GatherEnvironment) > 0 {
+		info, err := inspectWrapper(d.client, ctx, container.ID)
+		if err != nil {
+			return fmt.Errorf("Error getting docker container inspect: %s", err.Error())
+		}
+
+		for _, envvar := range info.Config.Env {
+			kvs := strings.SplitN(envvar, "=", 2)
+			if !sliceContains(kvs[0], d.GatherEnvironment) {
```

Comment:
Don't use this function since it is private to the libraries docker package, its just a coincidence that we are also in a docker package.  I think this loop would be better if we ranged over d.GatherEnvironment and split only if the prefix matches.